extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/internal/protocol/Products.java

```diff
@@ -74,15 +91,24 @@ public boolean isMultiZone() {
         return multiZone;
     }
 
-    public static Products getProductFromProductID(long id) throws IllegalArgumentException {
+    public static Products getLikelyProduct(ThingTypeUID uid) {
+        for (Products product : Products.values()) {
+            if (product.getThingTypeUID().equals(uid)) {
+                return product;
+            }
+        }
 
-        for (Products c : Products.values()) {
-            if (c.productID == id) {
-                return c;
+        throw new IllegalArgumentException(uid + " is not a valid product thing type UID");
+    }
+
+    public static Products getProductFromProductID(long id) throws IllegalArgumentException {
```

Comment:
Can you add a JavaDoc, that the return value if non-null all the time or a IllegalArgumentException is thrown?